src/main/scala/play/api/cache/redis/connector/RedisConnectorImpl.scala

```diff
@@ -294,6 +294,57 @@ private[connector] class RedisConnectorImpl(serializer: AkkaSerializer, redis: R
     }
   }
 
+  def zsetAdd(key: String, scoreValues: (Double, Any)*) = {
+    // encodes the value
+    def toEncoded(value: Any) = encode(key, value)
+    Future.sequence(scoreValues.map(scoreValue => toEncoded(scoreValue._2).map(encodedString => (scoreValue._1, encodedString)))).flatMap(redis.zadd(key, _: _*)) executing "ZADD" withKey key andParameters scoreValues expects {
```

Comment:
```suggestion
    def toEncoded(score: Double, member: Any) = encode(key, member).map(score -> _)
    Future.sequence(scoreValues.map((toEncoded _).tupled)).flatMap(redis.zadd(key, _: _*)) executing "ZADD" withKey key andParameters scoreValues expects {
```

something like this, might need to polish a bit, I didn't check it with compiler

Reply:
This code have been simplified already.